monai/transforms/croppad/array.py

```diff
@@ -1344,8 +1345,17 @@ def randomize(
             _shape = image.peek_pending_shape() if isinstance(image, MetaTensor) else image.shape[1:]
         if _shape is None:
             raise ValueError("label or image must be provided to infer the output spatial shape.")
+        img_shape = img.peek_pending_shape() if isinstance(image, MetaTensor) else img.shape[1:]
```

Comment:
```suggestion
        img_shape = img.peek_pending_shape() if isinstance(img, MetaTensor) else img.shape[1:]
```

I think this is what you meant, you may have to raise an exception if `img_shape` is None as well.

Reply:
Thanks,
I added a condition but if I raise an exception I am afraid it could break existing functionnalities, that is why I added this warning [here](https://github.com/Project-MONAI/MONAI/pull/7255/files/932e495d8d9103cd5bbeef67c30439e1a14e3249#diff-96cd15307195a8d855e3fb33c35abaa18afaecc182874751af4eb3144726d158R621), wdyt ?

Reply:
You can leave it in which case `generate_label_classes_crop_centers` will issue a warning every time when doing it's own check. That may be fine though I'm not sure under what conditions `img_shape` would be None and result in an exception further down stream and when it would just be a warning.